packages/eslint-plugin-choreography-ts/src/no-outside-choreographic-operator.ts

```diff
@@ -0,0 +1,184 @@
+              // know where to place the missing operator
+              let propertyRange: [number, number];
+              // find matching operator in the dependencies parameter
+              properties.forEach((prop) => {
+                if (prop.type === AST_NODE_TYPES.Property) {
+                  // Store the range of the current operator
+                  propertyRange = prop.range;
+                  if (prop.key.type === AST_NODE_TYPES.Identifier) {
+                    // This is more readable without using "&&"
+                    if (prop.key.name === operator) {
+                      match = true;
+                    }
+                  }
+                }
+              });
+              // if operator not in the dependencies parameter
+              if (!match) {
+                // fix to insert missing operator into the dependencies parameter
+                const fix = (fixer: TSESLint.RuleFixer) => {
+                  // Insert the missing operator after the last present operator
+                  return fixer.insertTextAfterRange(
+                    propertyRange,
+                    `, ${operator}`
+                  );
+                };
+                // report error and fixes
+                context.report({
+                  node: node,
+                  messageId: "error",
+                  data: {
+                    operator: operator,
+                  },
+                  suggest: [
+                    {
+                      messageId: "suggestion",
+                      fix: fix, // suggested fix (appears in list of suggestions)
+                    },
+                  ],
+                  fix: fix, // autofix (can be applied with `--fix`)
+                });
+              }
+              // otherwise if the dependencies object is empty
+            } else {
+              // fix to replace the empty object with one containing the missing operator
+              const fix = (fixture: TSESLint.RuleFixer) => {
+                return fixture.replaceTextRange(param.range, `{ ${operator} }`);
+              };
+              // report error and fixes
+              context.report({
+                node: node,
+                messageId: "error",
+                data: {
+                  operator: operator,
+                },
+                suggest: [
+                  {
+                    messageId: "suggestion",
+                    fix: fix,
+                  },
+                ],
+                fix: fix,
+              });
+            }
+            // otherwise if the dependencies object isn't present in the parameters
+            // but there are other parameter(s) present
+          } else {
+            // fix to insert missing dependencies parameter with missing operator
+            const fix = (fixer: TSESLint.RuleFixer) => {
+              // Insert the missing dependencies before the first parameter
```

Comment:
Oh, well I'm not renaming the missing operator when I insert it. I'm inserting it in the shorthand format `{ locally, colocally, ...}`.

Reply:
No, I understand. But having a normal variable (not object destructing syntax) already is a violation of `no-renaming-operator`. 

https://github.com/shumbo/choreography-ts/blob/4f88aad4a2cfc04475cf1f33415cb13aae9c35c0/packages/eslint-plugin-choreography-ts/tests/no-outside-choreographic-operator.test.ts#L98

For example, this is a violation of `no-renaming-operator`. I'm not sure if `no-outside-choreographic-operator` should push the first argument (`arg`) to the second because that would change the meaning of the code.

Reply:
Oh wait, I guess `no-renaming-operator` doesn't work on `colocally` although the second argument of `colocally` is `Choreography`. We might want to fix that later.

Reply:
> No, I understand. But having a normal variable (not object destructing syntax) already is a violation of `no-renaming-operator`.
> 
> https://github.com/shumbo/choreography-ts/blob/4f88aad4a2cfc04475cf1f33415cb13aae9c35c0/packages/eslint-plugin-choreography-ts/tests/no-outside-choreographic-operator.test.ts#L98
> 
> For example, this is a violation of `no-renaming-operator`. I'm not sure if `no-outside-choreographic-operator` should push the first argument (`arg`) to the second because that would change the meaning of the code.

Ok, so do you think it should just replace that argument then?

Reply:
And yeah, `no-renaming-operator` doesn't work for that. I can try fixing that too, although it seems like it might be a bit tougher since it isn't explicitly designated as being a choreography. I'll see what I can do

Reply:
@ImpregnableProgrammer 

> Ok, so do you think it should just replace that argument then instead of pushing it over?

I still think this problem is not something the linter can fix automatically. For me, replacing the argument would make more sense but still, it might break the user's code. I think the fix should only be implemented for errors that can automatically be solved 100%.

> I can try fixing that too, although it seems like it might be a bit tougher since it isn't explicitly designated as being a choreography

Let's work on this on a separate branch. Fixing this shouldn't be too bad. You can basically treat the second argument of `colocally` as a choreography, and the rest should be the same.

Reply:
> Fixing this shouldn't be too bad. You can basically treat the second argument of `colocally` as a choreography, and the rest should be the same.
Ok, although I suppose that would apply to all operators, and not just `colocally`.

Reply:
You're right. It's not all operators but `colocally` and `call`. Those two operators take a `Choreography` as an operand.